### ISSUE TITLE
ng lint warnings cannot read property of undefined

### ISSUE THREAD

**Issue by Martin-Wegner** [angular/angular-cli#5585]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report
- [ ] feature request
```

### Versions.
@angular/cli: 1.0.0-rc.4
node: 7.7.3
os: linux x64
@angular/common: 2.4.10
@angular/compiler: 2.4.10
@angular/core: 2.4.10
@angular/forms: 2.4.10
@angular/http: 2.4.10
@angular/platform-browser: 2.4.10
@angular/platform-browser-dynamic: 2.4.10
@angular/router: 3.4.10
@angular/cli: 1.0.0-rc.4
@angular/compiler-cli: 2.4.10

### Repro steps.
ng new test123
cd test123
ng lint

-> all ok

add `"no-unnecessary-qualifier": true` and `"promise-function-async": true` to `tslint.json`

ng lint --type-check

### The log given by the failure.
```
Warning: Cannot read property 'exports' of undefined
Warning: Cannot read property 'flags' of undefined
Warning: Cannot read property 'members' of undefined
```

### Desired functionality.
No warnings.

### Mention any other details that might be useful.
When I use webpack 2.3.1, tslint 4.5.1 and codelyzer 2.0.1 without angular-cli I get no warnings with this rules.

**Comment by filipesilva** [comment]
Sounds like a bug alright, thanks for the report!

**Comment by trusktr** [comment]
I get this warning when using tslint-loader in a Webpack project, but there's no Angular involved. Seems harmless. At least I haven't noticed what effect it has, if any.

**Comment by pgrm** [comment]
`ng lint` seems to work, just `ng lint --type-check` shows those errors. Also when I run tslint directly (even with `--type-check` I don't get those warnings

**Comment by sebek64** [comment]
I see

Warning: Cannot read property 'some' of undefined

when enabling no-unnecessary-qualifier rule. I'm running tslint directly. tslint 5.1.0, codelyzer 3.0.1. Maybe it is related to this one.

**Comment by sumitarora** [comment]
Tried to replicate on new project seems to be working fine. with below configuration.
```
@angular/cli: 1.1.0-beta.0
node: 7.5.0
```
devDependencies
```
    "@angular/cli": "1.1.0-beta.0",
    "@angular/compiler-cli": "^4.0.0",
    "@angular/language-service": "^4.0.0",
    "@types/jasmine": "2.5.38",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.0.0",
    "jasmine-core": "~2.5.2",
    "jasmine-spec-reporter": "~3.2.0",
    "karma": "~1.4.1",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "karma-coverage-istanbul-reporter": "^1.0.0",
    "protractor": "~5.1.0",
    "ts-node": "~2.0.0",
    "tslint": "~5.1.0",
    "typescript": "~2.3.1"
```

**Comment by sebek64** [comment]
I'm also confirming that I don't see the (maybe-related) issue with no-unnecessary-qualifier. I don't remember exactly if it was fixed by upgrading tslint to 5.2, but it was some component upgrade that fixed this warning.

**Comment by Martin-Wegner** [comment]
Yes it seems bo be fixed.
